browser/src/control/jsdialog/Util.FocusCycle.js

```diff
@@ -63,5 +63,63 @@ function makeFocusCycle(container, failedToFindFocusFunc) {
 	});
 }
 
+function findFocusableElement(element, direction) {
+
+	// check if no element present the just return null
+	if (!element) return null;
+
+	// Check the current element if it is focusable
+	if (isFocusable(element)) return element;
+
+	// Check if sibling is focusable or contains focusable elements
+	const focusableInSibling = findFocusableWithin(
+		element,
+		direction,
+	);
+	if (focusableInSibling) return focusableInSibling;
+
+	// Depending on the direction, find the next or previous sibling
+	const sibling =
+		direction === 'next'
+			? element.nextElementSibling
+			: element.previousElementSibling;
+
+	if (sibling) {
+		// Recursively check the next or previous sibling of the current sibling
+		return findFocusableElement(sibling, direction);
+	}
+
+	return null;
+}
+
+// Utility function to check if an element is focusable
+// This function diffrent from getFocusableElements. This only checkes the current element is focusable or not.
+function isFocusable(element) {
```

Comment:
can we put that function close to the other function which generates list of focusable elements inside container?
so we will not forget to update it in case we will modify the other, we should think also how to share the conditions between them in the future, if that possible (not here, in the followup)

Reply:
sure let me change the position